examples/bri-3/src/bri/decorators/public-key.ts

```diff
@@ -0,0 +1,12 @@
+import { createParamDecorator, ExecutionContext } from '@nestjs/common';
+import { decodeJWT } from 'did-jwt';
+
+export const PublicKey = createParamDecorator(
+  (data: unknown, ctx: ExecutionContext) => {
+    const request = ctx.switchToHttp().getRequest();
+    const bearerToken = request.headers['authorization'];
+    const token = bearerToken.split(' ')[1];
+    const publicKey = decodeJWT(token).payload.sub.split(':')[3];
```

Comment:
can token be undefined in worst case?

Reply:
@Manik-Jain I don't see a way for the token to be undefined if this decorator is used with a guarded route, but will add a thrown error just in the decorator is used improperly.

Reply:
I ended up doing the check on bearer token in case someone places this decorator on a public path. that is the only way they will reach this point without an authorization header. If the bearer token exists (even in an incorrect form) split should still return something defined even if delimiter is not found. Incorrect forms will be caught on the next line within the decodeJWT when identified as incorrect format.